test/definitions/spirv.jl

```diff
@@ -8,7 +8,8 @@ end
 # create a SPIRV-based test compiler, and generate reflection methods for it
 
 function spirv_job(@nospecialize(func), @nospecialize(types); kernel::Bool=false, kwargs...)
-    source = FunctionSpec(func, Base.to_tuple_type(types), kernel)
+    f_type = isa(func, Type) ? Type{func} : typeof(func)
```

Comment:
maybe that's what we should do in the constructor for `FunctionSpec`?

Reply:
I originally had this in the native, but it hit the closure ambiguity so I removed it from there.